### DISCUSSION TITLE
CNAS AIR-09: Support

### DISCUSSION THREAD

**Discussion by jordanch68** [Gissio/radpro#100] [Ideas]
Hello, I'd like to request Rad Pro firmware support for the AIR-09 geiger counter. The factory firmware is broken, it does not provide results in other units besides uSv even though it can be changed in the settings. 

I can provide other details if needed, I think the firmware update process would be similar to the Bosean FS-5000 but that's just a guess. It came with a J305 tube and I measured the HV at 443V, this can also use a J321 tube and I tried that with the same results. It works, just needs better firmware. I've applied the Rad Pro firmware on my Bosean and GC-01 so I'm familiar with the process. 

Thank you,
William C Jordan
![17183391830201511876781412143985](https://github.com/Gissio/radpro/assets/172728764/0e6c734b-5640-4939-8726-cf51ede19dd3)

**Comment by Gissio** [comment]
I currently lack the time and motivation for what you propose.

This is what would need to be done:

* The original firmware of the device needs to be dumped, disassembled and analyzed to understand how the microcontroller interacts with its internal devices (input/output ports, timers, ADCs, etc).
* A schematics of the device needs to be created by tracing the connections from the microcontroller to the various components of the electronics board.
* Rad Pro's code needs to be customized for this device.
